community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/sampling/IndexSamplingConfig.java

```diff
@@ -30,9 +30,17 @@
 
     public IndexSamplingConfig( Config config )
     {
-        this.sampleSizeLimit = config.get( GraphDatabaseSettings.index_sample_size_limit );
-        this.updateRatio = ((double) config.get( GraphDatabaseSettings.index_sampling_update_percentage )) / 100.0d;
-        this.backgroundSampling = config.get( GraphDatabaseSettings.index_background_sampling_enabled );
+        this( config.get( GraphDatabaseSettings.index_sample_size_limit ),
+              config.get( GraphDatabaseSettings.index_sampling_update_percentage ) / 100.0d,
+              config.get( GraphDatabaseSettings.index_background_sampling_enabled )
+        );
+    }
+
+    public IndexSamplingConfig( int sampleSizeLimit, double updateRatio, boolean backgroundSampling )
```

Comment:
This new constructor seems only to be used from the one above. What was the reason for splitting it up?

Reply:
I think it was added to help make some testing easier, but it seems we reverted that part without reverting this. I'll remove it while rebasing.